### DISCUSSION TITLE
Version 2 guides and timeline for new users

### DISCUSSION THREAD

**Discussion by nsheff** [sqlalchemy/sqlalchemy#8135] [Usage Questions]
Hey, I have a question about the upcoming version 2 release. Does anyone know if there's an ETA or timeline for when the target is?

The reason I ask is that I'm a new user, who hasn't used sqlalchemy before, and I'm interested in using it for a project... but I don't really want to wade through learning the "old way" of doing things only to have to change shortly thereafter, I'd rather just start fresh with the version 2.0 docs/syntax/ideas. So I've been holding off on starting this project, waiting for version 2.0 release since I saw that it was in progress. I see lots of documentation on how to migrate from 1.X to 2.0, and how 2.0 compares to 1.X, etc, but this isn't really useful to me since I'm unfamiliar with 1.X. I was hoping to just be able to follow the *new* tutorial that just explains 2.0 without the history/cruft of previous concepts. So, I guess that kind of leads to a second question: is there anywhere I can go to learn 2.0 without having to first familiarize myself with 1.X ideas that are going away?

Thanks.

**Comment by CaselIT** [comment]
> Hey, I have a question about the upcoming version 2 release. Does anyone know if there's an ETA or timeline for when the target is?

There is not firm eta at the moment, the idea is to get initial beta out by the end of the year.



> So, I guess that kind of leads to a second question: is there anywhere I can go to learn 2.0 without having to first familiarize myself with 1.X ideas that are going away?

Version 1.4 mostly support v2 syntax. The unified tutorial is the best place to start https://docs.sqlalchemy.org/en/14/tutorial/index.html
The main change at the moment from 1.4 to v2 is how orm mapping is defined to fully support pep484. Other than that the api is mostly the same. A preview of the new style can be seen here https://docs.sqlalchemy.org/en/20/orm/quickstart.html

**Reply by zzzeek** [reply]
agree, I think read the current 1.4 tutorial and you'll have plenty to learn that is fully transferrable to 2.0.  The main thing that will "feel" different in 2.0 is the availability of slightly different constructs that will get you better IDE integration and typing support.   so I would just not get too deeply into pep-484 typing with 1.4.x, like wrestling with the stubs, etc.  that's the main thing that will have a difference in 2.0 vs. 1.4.
